platform/linux-generic/odp_ishm.c

```diff
@@ -1094,12 +1111,15 @@ int _odp_ishm_reserve(const char *name, uint64_t size, int fd,
 
 	new_block = &ishm_tbl->block[new_index];
 
-	/* save block name (if any given): */
-	if (name)
-		_odp_strcpy(new_block->name, name, ISHM_NAME_MAXLEN);
-	else
+	/* Save application given block name as-is and make internal block name used for mmap
+	   unique */
+	if (name) {
```

Comment:
Maybe it would be simpler to always make the internal name unique, even if no name was given by the application? Now the situation is a bit weird that the internal name is unique except when it is the empty string in which case it is not unique and then the lower level code has to special case that and create a unique string using create_seq_string(). Making the name always unique would make the code more clear, I think.

Reply:
Logic changed in V3 to always use unique internal names.